includes/specials/SpecialUserlogout.php

```diff
@@ -92,6 +92,14 @@ function execute( $par ) {
 			}
 		}
 
-		$out->returnToMain(false, $mReturnTo, $mReturnToQuery);
+		if (
+			empty( $mReturnTo ) &&
+			empty( $mReturnToQuery ) &&
+			$redirectUrl = $this->getRequest()->getVal( 'redirect' )
+		) {
+			$out->redirect( $redirectUrl );
```

Comment:
Are there any constraints for `$redirectUrl`, (e.g. will always be at a fandom.wikia.com host)?  One reason `returnTo` was limited to just wiki pages was probably to keep people from using Wikia as a mask for some h4x0r URL that they want to masquerade as a safe wikia URL.  If we could do some checking on this URL it would be good.